File: src/onion/compiler/Modifier.java
package onion.compiler;
/* ************************************************************** *
* *
* Copyright (c) 2005, Kota Mizushima, All rights reserved. *
* *
* *
* This software is distributed under the modified BSD License. *
* ************************************************************** */

/**
* @author Kota Mizushima
* Date: 2005/04/10
*/
public class Modifier {
  public static final int INTERNAL = 1;
  public static final int SYNCHRONIZED = 2;
  public static final int FINAL = 4;
  public static final int ABSTRACT = 8;
  public static final int VOLATILE = 16;
  public static final int STATIC = 32;
  public static final int INHERITED = 64;
  public static final int PUBLIC = 128;
  public static final int PROTECTED = 256;
  public static final int PRIVATE = 512;
  public static final int FORWARDED = 1024;

  public static boolean check(int modifier, int bitFlag){
    return (modifier & bitFlag) != 0;
  }

  public static boolean isInternal(int modifier){
    return check(modifier, INTERNAL);
  }

  public static boolean isStatic(int modifier){
    return check(modifier, STATIC);
  }

  public static boolean isSynchronized(int modifier){
    return check(modifier, SYNCHRONIZED);
  }

  public static boolean isFinal(int modifier){
    return check(modifier, FINAL);
  }

  public static boolean isAbstract(int modifier){
    return check(modifier, ABSTRACT);
  }

  public static boolean isVolatile(int modifier){
    return check(modifier, VOLATILE);
  }

  public static boolean isInherited(int modifier){
    return check(modifier, INHERITED);
  }

  public static boolean isPublic(int modifier){
    return check(modifier, PUBLIC);
  }

  public static boolean isProtected(int modifier){
    return check(modifier, PROTECTED);
  }

  public static boolean isPrivate(int modifier){
    return check(modifier, PRIVATE);
  }

  public static boolean isForwarded(int modifier){
    return check(modifier, FORWARDED);
  }

}

File: src/main/scala/onion/compiler/Modifier.scala
package onion.compiler;
/* ************************************************************** *
* *
* Copyright (c) 2016-, Kota Mizushima, All rights reserved. *
* *
* *
* This software is distributed under the modified BSD License. *
* ************************************************************** */

/**
* @author Kota Mizushima
*
*/
object Modifier {

  final val INTERNAL = 1
  final val SYNCHRONIZED = 2
  final val FINAL = 4
  final val ABSTRACT = 8
  final val VOLATILE = 16
  final val STATIC = 32
  final val INHERITED = 64
  final val PUBLIC = 128
  final val PROTECTED = 256
  final val PRIVATE = 512
  final val FORWARDED = 1024
  final val SEALED = 2048
  final val ENUM = 4096

  def check(modifier: Int, bitFlag: Int): Boolean = {
     (modifier & bitFlag) != 0
  }

  def isInternal(modifier: Int): Boolean = {
     check(modifier, INTERNAL)
  }

  def isStatic(modifier: Int): Boolean = {
     check(modifier, STATIC)
  }

  def isSynchronized(modifier: Int): Boolean = {
     check(modifier, SYNCHRONIZED)
  }

  def isFinal(modifier: Int): Boolean = {
     check(modifier, FINAL)
  }

  def isAbstract(modifier: Int): Boolean = {
     check(modifier, ABSTRACT)
  }

  def isVolatile(modifier: Int): Boolean = {
     check(modifier, VOLATILE)
  }

  def isInherited(modifier: Int): Boolean = {
     check(modifier, INHERITED)
  }

  def isPublic(modifier: Int): Boolean = {
     check(modifier, PUBLIC)
  }

  def isProtected(modifier: Int): Boolean = {
     check(modifier, PROTECTED)
  }

  def isPrivate(modifier: Int): Boolean = {
     check(modifier, PRIVATE)
  }

  def isForwarded(modifier: Int): Boolean = {
     check(modifier, FORWARDED)
  }

  def isSealed(modifier: Int): Boolean = {
     check(modifier, SEALED)
  }

  def isEnum(modifier: Int): Boolean = {
     check(modifier, ENUM)
  }

}
